packages/langium/test/parser/indentation-aware.test.ts

```diff
@@ -181,69 +194,100 @@ describe('indentationAwareLexer', () => {
-    hidden terminal NL: /[\\r\\n]+/;
-    hidden terminal WS: /[\\t ]+/;
+        terminal BOOLEAN returns boolean: /true|false/;
+        terminal INDENT: 'synthetic:indent';
+        terminal DEDENT: 'synthetic:dedent';
+        hidden terminal NL: /[\\r\\n]+/;
+        hidden terminal WS: /[\\t ]+/;
     `;
 
     test('should parse correctly indented code', async () => {
         const parser = await getParser(sampleGrammar);
-        const { parserErrors } = parser.parse(`
-if true:
-    return false
-else:
-    return true`);
+        const { parserErrors } = parser.parse(expandToString`
+        if true:
+            return false
+        else:
+            return true
+        `);
 
         expect(parserErrors).toHaveLength(0);
     });
 
     test('should error on non-matching dedent', async () => {
         const parser = await getParser(sampleGrammar);
-        const {parserErrors} = parser.parse(`
-if true:
-    return false
-  else:
-    return true`);
+        const { parserErrors } = parser.parse(expandToString`
+        if true:
+            return false
+          else:
+            return true
+        `);
 
         expect(parserErrors.length).toBeGreaterThan(0);
     });
 
     test('should throw an error on unexpected indent', async () => {
         const parser = await getParser(sampleGrammar);
-        const { parserErrors } = parser.parse(`
-        if true:
-            return false`);
+        const { parserErrors } = parser.parse(expandToString`
+        // Parsing starts here
+                if true:
+                    return false
+        `);
 
         expect(parserErrors.length).toBeGreaterThan(0);
     });
 
     test('should correctly parse nested blocks', async () => {
         const parser = await getParser(sampleGrammar);
-        const {parserErrors} = parser.parse(`
-if true:
-    return true
-else:
-    if false:
-        return false
+        const { parserErrors, value } = parser.parse(expandToString`
+        if true:
+            return true
+        else:
+            if false:
+                return true
+                return false
+            return true
         `);
 
         expect(parserErrors).toHaveLength(0);
+        const ifValue = value as If;
+        expect(ifValue.thenBlock).toHaveLength(1);
+        expect(ifValue.elseBlock).toHaveLength(2);
+        const elseBlock = ifValue.elseBlock![0] as If;
+        expect(elseBlock.thenBlock).toHaveLength(2);
+        const nestedReturn1 = elseBlock.thenBlock[0] as Return;
+        expect(nestedReturn1.value).toBe(true);
+        const nestedReturn2 = elseBlock.thenBlock[1] as Return;
+        expect(nestedReturn2.value).toBe(false);
+        const return2 = ifValue.elseBlock![1] as Return;
+        expect(return2.value).toBe(true);
     });
 
 });
+
+type Statement = If | Return;
+
+interface If extends AstNode {
+    $type: 'If';
+    condition: boolean;
+    thenBlock: Statement[];
+    elseBlock?: Statement[];
```

Comment:
Why is this field optional? Since it uses `+=`, shouldn't there always be an array (even if empty) by default?

Reply:
GitHub Copilot strikes again. Yes, you're right :D